pulsar-client-cpp/lib/MultiTopicsConsumerImpl.cc

```diff
@@ -711,3 +736,81 @@ uint64_t MultiTopicsConsumerImpl::getNumberOfConnectedConsumer() {
     });
     return numberOfConnectedConsumer;
 }
+void MultiTopicsConsumerImpl::runPartitionUpdateTask() {
+    partitionsUpdateTimer_->expires_from_now(partitionsUpdateInterval_);
+    partitionsUpdateTimer_->async_wait([this](const boost::system::error_code& ec) {
```

Comment:
You must capture `shares_from_this()` here, otherwise `this` might point to a destroyed object and `topicPartitionUpdate()` (short for `this->topicPartitionUpdate()`) would cause segmentation fault.